requirements/adjust_versions.py

```diff
@@ -1,47 +1,50 @@
 import os
 import re
 import sys
-from typing import Any, Dict
-
-VERSIONS_LUT: Dict[str, Dict[str, Any]] = {
-    "1.4.0": dict(torchvision="0.5.0", torchtext="0.5"),
-    "1.5.0": dict(torchvision="0.6.0", torchtext="0.6"),
-    "1.5.1": dict(torchvision="0.6.1", torchtext="0.6"),
-    "1.6.0": dict(torchvision="0.7.0", torchtext="0.7"),
-    "1.7.0": dict(torchvision="0.8.1", torchtext="0.8"),
-    "1.7.1": dict(torchvision="0.8.2", torchtext="0.8.1"),
-    "1.8.0": dict(torchvision="0.9.0", torchtext="0.9"),
-    "1.8.1": dict(torchvision="0.9.0", torchtext="0.9"),
-}
-
-
-def find_latest(ver: str, versions_all: list) -> str:
+from typing import Dict, Optional
+
+# IMPORTANT: this list must be sorted from latest to oldest
+VERSIONS = [
+    dict(torch="1.9.0", torchvision="", torchtext=""),  # nightly
+    dict(torch="1.8.1", torchvision="0.9.1", torchtext="0.9.1"),
+    dict(torch="1.8.0", torchvision="0.9.0", torchtext="0.9.0"),
+    dict(torch="1.7.1", torchvision="0.8.2", torchtext="0.8.1"),
+    dict(torch="1.7.0", torchvision="0.8.1", torchtext="0.8.0"),
+    dict(torch="1.6.0", torchvision="0.7.0", torchtext="0.7"),
+    dict(torch="1.5.1", torchvision="0.6.1", torchtext="0.6"),
+    dict(torch="1.5.0", torchvision="0.6.0", torchtext="0.6"),
+    dict(torch="1.4.0", torchvision="0.5.0", torchtext="0.5"),
+]
+
+
+def find_latest(ver: str) -> Dict[str, str]:
     # drop all except semantic version
     ver = re.search(r'([\.\d]+)', ver).groups()[0]
-    # find candidates, by starting version pattern
-    options = [v for v in versions_all if v.startswith(ver)]
-    assert options, f"missing {ver} among {versions_all}"
-    # take the last one...
-    return sorted(options)[-1]
 
+    # find first match
+    for option in VERSIONS:
+        if option["torch"].startswith(ver):
+            return option
 
-def main(path_req: str, torch_version: str = None) -> None:
-    with open(path_req, "r") as fp:
-        req = fp.read()
+    raise ValueError(f"Missing {ver} in {VERSIONS}")
 
+
+def main(path_req: str, torch_version: Optional[str] = None) -> None:
     if not torch_version:
         import torch
         torch_version = torch.__version__
-    assert torch_version, f"invalid/missing Torch: {torch_version}"
-
-    torch_version = find_latest(torch_version, list(VERSIONS_LUT.keys()))
-    dep_versions = VERSIONS_LUT[torch_version]
-    dep_versions["torch"] = torch_version
-    for lib in dep_versions:
-        version = dep_versions[lib]
-        replace = f"{lib}=={version}\n"
+    assert torch_version, f"invalid torch: {torch_version}"
+
+    with open(path_req, "r") as fp:
+        req = fp.read()
+
+    latest = find_latest(torch_version)
+    for lib, version in latest.items():
+        replace = f"{lib}=={version}" if version else lib
+        replace += os.linesep
         req = re.sub(rf"{lib}[>=]*[\d\.]*{os.linesep}", replace, req)
 
+    print(req)
```

Comment:
Do we need to print ?

Reply:
It's helpful as this is used in CI and the print shows the changes to the file.

But we don't *need* it